### PR TITLE
Change sales response to data

### LINKED ISSUE #100: Return Response instead of Void for "Sales and Finance Reports" endpoints
I try to get finance reports from API. To get the gzip archive I need access to ```Response.data``` which is not provided for ```APIEndpoint```s with ```T == Void```.

Is it possible to provide the ```Response``` object instead of ```Void``` to the completion handler?

### PR DESCRIPTION
**Pull request by DechengMa**
Closes #100 

The Apple Store Connect API `/v1/financeReports` and `/v1/salesReports` will return binary `a-gzip` data which is not supported by the SDK yet. 
This PR is to add support for downloading sales and trends reports.

## Changes 
- downloadSalesAndTrendsReports, downloadFinanceReports `APIEndpoint: T == Void` -> `Data`
- Tweak APIProvider for return raw data if `T.self == Data.self`

### CODE REVIEW

File: Sources/APIProvider.swift [hunk 1]
```diff
@@ -163,6 +163,10 @@ private extension APIProvider {
                 return .failure(Error.requestFailure(response.statusCode, response.data))
             }
 
+            if T.self is Data.Type {
+                return .success(data as! T)
```
Comment (AvdLee) [inline Sources/APIProvider.swift:167]: This is a bit nicer and removes the force unwrap!

```suggestion
            if let data as? T {
                return .success(data)
```

Comment (DechengMa) [reply Sources/APIProvider.swift:167]: @AvdLee Thanks for the reply, I have tried this but it gave me a syntax error and couldn't compile, did you try this before?

Comment (kinnarr) [reply Sources/APIProvider.swift:167]: `let data = data as? T` could work how @AvdLee expected the suggestion would work

Comment (DechengMa) [reply Sources/APIProvider.swift]: Thanks mate, it works

**Comment by AvdLee** [comment]
Thanks for adding the test! 🎉

File: Sources/APIProvider.swift (hunk 1 above)
Comment (AvdLee) [reply Sources/APIProvider.swift]: Oops, my bad. Thanks for correcting me @kinnarr!

**Pull request merged**